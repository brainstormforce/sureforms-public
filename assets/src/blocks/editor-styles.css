.sureform-add-option-container {
	display: flex;
	gap: 1rem;
	align-items: flex-start;
	justify-content: center;
}

.sureform-add-option-button {
	height: 30px;
}

.sureform-add-option-container .components-base-control__field {
	padding-top: 0px !important;
}
.sureforms-select__input:focus{
	box-shadow: none !important;
}
