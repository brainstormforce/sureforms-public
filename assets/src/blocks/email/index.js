/**
 * WordPress dependencies
 */
import { MdOutlineMail as icon } from 'react-icons/md';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';

const { name } = metadata;

export { metadata, name };

export const settings = {
	icon,
	edit,
};
