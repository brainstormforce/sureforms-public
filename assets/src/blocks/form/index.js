/**
 * WordPress dependencies
 */
import icon from '../../../../images/Logo.js';

/**
 * Internal dependencies
 */
import edit from './edit';
import save from './save';
import metadata from './block.json';

const { name } = metadata;

export { metadata, name };

export const settings = {
	icon,
	edit,
	save,
};
