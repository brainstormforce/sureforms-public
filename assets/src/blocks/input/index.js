/**
 * WordPress dependencies
 */
import { MdOutlineTextFields as icon } from 'react-icons/md';

/**
 * Internal dependencies
 */
import edit from './edit';
import save from './save';
import metadata from '../../../../inc/blocks/input/block.json';

const { name } = metadata;

export { metadata, name };

export const settings = {
	icon,
	edit,
	save,
};
