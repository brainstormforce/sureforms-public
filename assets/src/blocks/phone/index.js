/**
 * WordPress dependencies
 */
import { mobile as icon } from '@wordpress/icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import save from './save';
import metadata from './block.json';

const { name } = metadata;

export { metadata, name };

export const settings = {
	icon,
	edit,
	save,
};
