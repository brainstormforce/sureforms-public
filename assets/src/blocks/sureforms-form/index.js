/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { quote as icon } from '@wordpress/icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import save from './save';
import metadata from './block.json';

export { metadata };
export const settings = {
	icon,
	edit,
	save,
};
