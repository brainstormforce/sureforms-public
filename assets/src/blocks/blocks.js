import * as sfForm from '@Blocks/sureforms-form';
import * as form from '@Blocks/form';
import * as text from '@Blocks/input';
import * as number from '@Blocks/number';
import * as email from '@Blocks/email';
import * as textarea from '@Blocks/textarea';
import * as checkbox from '@Blocks/checkbox';
import * as radioGroup from '@Blocks/radio-group';
import * as radio from '@Blocks/radio';
import * as toggle from '@Blocks/switch';
import * as submit from '@Blocks/submit';
import * as select from '@Blocks/dropdown';

import { registerBlocks } from './register-block';

registerBlocks( [
	form,
	text,
	email,
	textarea,
	radioGroup,
	radio,
	toggle,
	checkbox,
	number,
	sfForm,
	submit,
	select,
] );
