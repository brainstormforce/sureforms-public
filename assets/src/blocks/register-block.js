import { registerBlockType, createBlock } from '@wordpress/blocks';
import { getBlockTypes } from './util';

/**
 * Function to register blocks provided by SureForms.
 */
export const registerBlocks = (blocks = []) => {
	return (blocks || []).forEach(registerBlock);
};

/**
 * Function to register an individual block.
 *
 * @param {Object} block The block to be registered.
 *
 */
const registerBlock = (block) => {
	if (!block) {
		return;
	}

	const { metadata, settings } = block;

	const additionalSettings = 'sureforms/form' !== metadata.name ? {
		transforms: {
			from: [{
				type: 'block',
				blocks: getBlockTypes(metadata.name),
				transform: (attributes) => { return createBlock( metadata.name, attributes) },
			}],
		}
	} : {};

	registerBlockType(
		{
			...metadata,
			textdomain: 'sureforms', // set our text domain for everything.
		},
		{
			...settings,
			...additionalSettings,
		}
	);
};
