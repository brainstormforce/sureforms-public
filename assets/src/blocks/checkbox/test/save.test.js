/* eslint-disable no-undef */

/**
 * External dependencies
 */
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

describe( 'surecart/buttons', () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );
	it( 'Should render with defaults', () => {
		serializedBlock = serialize( block );
		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'wp:surecart/input' );
	} );

	it( 'should render with custom class name', () => {
		block.attributes.className = 'my-custom-class';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'my-custom-class' );
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
