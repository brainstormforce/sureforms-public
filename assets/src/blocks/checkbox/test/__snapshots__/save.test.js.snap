// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`surecart/buttons should render with custom class name 1`] = `"<!-- wp:surecart/input {\\"className\\":\\"my-custom-class\\"} /-->"`;
